modules/internal/ChapelArray.chpl

```diff
@@ -2293,7 +2295,17 @@ module ChapelArray {
              "  Actual domain is: ", this.domain);
     }
 
-    pragma "no doc"
+    /* Return an array declared over a new domain, provided that the new
+       domain is of the same rank and size as the original.
```

Comment:
This wording suggests to me that this routine creates a new array (with new storage associated with it), when in fact it returns a view of the original array, simply using new indices.